File: front-end/src/app/pages/sql-editor/Services/query-string/querying.service.ts
import {Inject, Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {API_PREFIX} from "../../../../app.token.d";
import {Observable} from "rxjs";
import {QueryResultVO} from "../vo/query-result.vo";

/**
 * @author Davood Akbari - 1399
 * daak1365@gmail.com
 * daak1365@yahoo.com
 * 09125188694
 */

@Injectable({
    providedIn: 'root'
})
export class QueryingService {
    private prefix: string = `${this.apiPrefix}/querying`;

    constructor(@Inject(API_PREFIX) private apiPrefix: string, private httpClient: HttpClient) {

    }

    public query(queryString: string): Observable<QueryResultVO> {
        return this.httpClient.post<QueryResultVO>(`${this.prefix}/query`, {query: queryString});
    }
}

File: back-end/src/main/java/ir/caspco/visualization/body/context/service/QueryingService.java
package ir.caspco.visualization.body.context.service;

import ir.caspco.visualization.body.context.vo.QueryResultVO;

import java.sql.SQLException;

/**
 * @author Davood Akbari - 1399
 * daak1365@gmail.com
 * daak1365@yahoo.com
 * 09125188694
 */

public interface QueryingService {
    QueryResultVO query(String queryIn) throws SQLException;
}
